### PR TITLE
Add NodeJS test for parameterized providers #2

### PR DESCRIPTION
**Pull request by julienp**
This takes https://github.com/pulumi/pulumi/pull/16392 and updates local SDK generation for nodejs and adds parametrisation support for invokes

### CODE REVIEW

File: sdk/nodejs/runtime/invoke.ts [hunk 1]
```diff
@@ -213,19 +213,29 @@ export class StreamInvokeResponse<T> implements AsyncIterable<T> {
...
+        opts.pluginDownloadURL = undefined;
+      }
+    }
     const req = new resourceproto.ResourceInvokeRequest();
     req.setTok(tok);
     req.setArgs(obj);
     req.setProvider(provider || "");
     req.setVersion(opts.version || "");
     req.setAcceptresources(!utils.disableResourceReferences);
+    req.setPlugindownloadurl(opts.pluginDownloadURL || "")
```
Comment (julienp) [inline sdk/nodejs/runtime/invoke.ts:237]: We didn't set `pluginDownloadURL` before for nodejs, which seems like a bug? For param providers this is cleared above.

File: tests/testdata/codegen/simplified-invokes/nodejs/absMultiArgs.ts [hunk 2]
```diff
@@ -29,5 +29,5 @@ export function absMultiArgsOutput(a: pulumi.Input<number>, b?: pulumi.Input<num
         "a": a,
         "b": b,
     };
-    return pulumi.output(args).apply((resolvedArgs: any) => absMultiArgs(resolvedArgs.a, resolvedArgs.b, opts))
+    return pulumi.output(args).apply((resolvedArgs: any) => absMultiArgs(resolvedArgs.a, resolvedArgs.b, opts));
```
Comment (Frassle) [inline tests/testdata/codegen/simplified-invokes/nodejs/absMultiArgs.ts:32]: Can this fix be pulled into it's own little PR?

File: sdk/nodejs/runtime/invoke.ts (hunk 1 above)
Comment (Frassle) [reply sdk/nodejs/runtime/invoke.ts]: This too probably worth just fixing alone. We ought to make sure we have a conformance test issue raised to cover this as well.

Comment (julienp) [reply sdk/nodejs/runtime/invoke.ts]: Yep, removed the change here and created an issue for me to follow up.

File: tests/testdata/codegen/simplified-invokes/nodejs/absMultiArgs.ts (hunk 2 above)
Comment (julienp) [reply tests/testdata/codegen/simplified-invokes/nodejs/absMultiArgs.ts:32]: Yep, will do

File: tests/integration/integration_nodejs_test.go [hunk 3]
```diff
@@ -2029,3 +2030,67 @@ func TestNodeOOM(t *testing.T) {
 		},
 	})
 }
+
+// Test a paramaterized provider with nodejs.
```
Comment (lunaris) [inline tests/integration/integration_nodejs_test.go:2034]:
```suggestion
// Test a parameterized provider with nodejs.
```

File: tests/integration/integration_nodejs_test.go [hunk 4]
```diff
@@ -2029,3 +2030,67 @@ func TestNodeOOM(t *testing.T) {
...
+//
+//nolint:paralleltest // mutates environment
+func TestParamaterizedNode(t *testing.T) {
```
Comment (lunaris) [inline tests/integration/integration_nodejs_test.go:2037]:
```suggestion
func TestParameterizedNode(t *testing.T) {
```

**Review by lunaris — APPROVED** [review]

File: pkg/codegen/nodejs/gen.go [hunk 5]
```diff
@@ -2397,14 +2432,38 @@ func genNPMPackageMetadata(pkg *schema.Package, info NodePackageInfo, localDepen
...
 	}
 
+	var pulumiPlugin plugin.PulumiPluginJSON
+	if pkg.Parameterization != nil {
+		pulumiPlugin = plugin.PulumiPluginJSON{
+			Resource: true,
+			Name:     pkg.Parameterization.BaseProvider.Name,
+			Version:  pkg.Parameterization.BaseProvider.Version.String(),
+			// TODO: Need PluginDownloadURL on BaseProvider
+			// Server: pkg.BaseProvider.PluginDownloadURL,
```
[the remark this answers is not in the document]
Comment (Frassle) [reply pkg/codegen/nodejs/gen.go:2451]: Oh.... we should fix this now. It's fixed in pkg/schema this line just needs uncommenting.

**Pull request merged**